### PR TITLE
[rush-lib] Add ability to opt out of changelog files for version policies.

### PR DESCRIPTION
**Pull request by lahuey**
Right now, all Rush projects that have a version policy will require changelogs files. This pull request adds the ability for a specific version policy to opt out of changelog verification.

### CODE REVIEW

File: apps/rush-lib/src/schemas/version-policies.schema.json [hunk 1]
```diff
@@ -83,10 +83,14 @@
...
           "type": "string"
+        },
+        "shouldSkipChangelogVerification": {
+          "description": "If true, the version policy will not require changelog files.",
+          "type": "boolean"
         }
       },
       "required": ["policyName", "definitionName", "version", "nextBump"],
-      "additionalProperties": false
+      "additionalProperties": true
```
Comment (iclanton) [inline apps/rush-lib/src/schemas/version-policies.schema.json]: You still want this to be `false`. From the [spec](https://json-schema.org/understanding-json-schema/reference/object.html):

>The `additionalProperties` keyword is used to control the handling of extra stuff, that is, properties whose names are not listed in the properties keyword. By default any additional `properties` are allowed.

File: apps/rush-lib/src/schemas/version-policies.schema.json [hunk 2]
```diff
@@ -105,10 +109,14 @@
...
           "type": "number"
+        },
+        "shouldSkipChangelogVerification": {
+          "description": "If true, the version policy will not require changelog files.",
+          "type": "boolean"
         }
       },
       "required": ["policyName", "definitionName"],
-      "additionalProperties": false
+      "additionalProperties": true
```
Comment (iclanton) [inline apps/rush-lib/src/schemas/version-policies.schema.json]: Same here.

File: apps/rush-lib/src/schemas/version-policies.schema.json [hunk 3]
```diff
@@ -105,10 +109,14 @@
         "lockedMajor": {
           "description": "The locked major version",
           "type": "number"
+        },
```
Comment (iclanton) [inline apps/rush-lib/src/schemas/version-policies.schema.json]: There is a somewhat wonky way to do inheritance in JSON schema. Perhaps you can dedupe this a little by doing that.

File: apps/rush-lib/src/api/VersionPolicy.ts [hunk 4]
```diff
@@ -107,6 +109,13 @@ export abstract class VersionPolicy {
     return this.definitionName === VersionPolicyDefinitionName.lockStepVersion;
   }
 
+  /**
+   * Determines if a version policy wants to opt out of changelog files.
+   */
+  public get shouldSkipChangelogVerification(): boolean {
```
Comment (iclanton) [inline apps/rush-lib/src/api/VersionPolicy.ts]: Include this in the init template: https://github.com/microsoft/web-build-tools/blob/master/apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json

File: apps/rush-lib/src/schemas/version-policies.schema.json [hunk 5]
```diff
@@ -83,10 +83,14 @@
         "mainProject": {
           "description": "The main project for this version policy",
           "type": "string"
+        },
+        "shouldSkipChangelogVerification": {
```
Comment (iclanton) [inline apps/rush-lib/src/schemas/version-policies.schema.json]: We should talk about this name. What do you think about `noChangeLogRequired` instead?

Is the behavior here just to not error when `rush change -v` is run, or is it to not prompt for changelogs at all?

Comment (lahuey) [reply apps/rush-lib/src/schemas/version-policies.schema.json]: The latter. NoChangeLogRequired works for me

File: apps/rush-lib/src/schemas/version-policies.schema.json (hunk 3 above)
Comment (lahuey) [reply apps/rush-lib/src/schemas/version-policies.schema.json]: Schema inheritance doesn't seem to support the "additionalProperties": false :(

File: common/changes/@microsoft/rush/master_2019-06-07-00-24.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "[rush-lib] Add ability to opt out of changelog files for version policies.",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/master_2019-06-07-00-24.json]: Remove the `[rush-lib] `. Changelogs are already attached to packages.

File: apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json [hunk 7]
```diff
@@ -77,7 +77,13 @@
...
      * to the types of changes made to each project, according to the "rush change" command.
      */
-    "lockedMajor": 3
+    "lockedMajor": 3,
+
+    /**
+     * (Optional) If false, developers of the specified version policy will no longer
+     * be prompted for changelog files during the rush change command.
+     */
+    "noChangeLogRequired": false
```
Comment (octogonz) [inline apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json]: Let's rename this to `exemptFromRushChange`.  It more clearly explains the effect, and avoids a double-negative `noIceCream = false` (*What!? Do I get 🍦 or not!!*)

File: apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json [hunk 8]
```diff
@@ -77,7 +77,13 @@
      * inappropriately.  The minor/patch version parts will be bumped independently according
      * to the types of changes made to each project, according to the "rush change" command.
      */
-    "lockedMajor": 3
+    "lockedMajor": 3,
+
+    /**
+     * (Optional) If false, developers of the specified version policy will no longer
+     * be prompted for changelog files during the rush change command.
```
Comment (octogonz) [inline apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json]: Incorrect wording:  `rush change` collects "change log entries" and makes "change files".  Then `rush version` integrates "change files" to produce a "change log" (i.e. CHANGELOG.md).

Comment (octogonz) [reply apps/rush-lib/assets/rush-init/common/config/rush/version-policies.json]: Maybe something like this:

"(Optional) When publishing is managed by Rush, by default the "rush change" command will request changes for any projects that are modified by a pull request.  These change entries will produce a CHANGELOG.md file.  If you author your CHANGELOG.md manually or announce updates in some other way, set "noChangeLogRequired" to true to tell "rush change" to ignore the projects belonging to this version policy"

**Pull request merged**